test/test_foreach.py

```diff
@@ -125,21 +111,79 @@ def _get_funcs(self, op):
             RegularFuncWrapper(op.ref_inplace),
         )
 
+    # note(crcrpar): Make sure 0-size tensors are appropriately ignored by `multi_tensor_apply`
+    # which is originally reported in https://github.com/pytorch/pytorch/issues/94865.
+    # rel:
+    #   - https://github.com/pytorch/pytorch/pull/94655
+    #   - https://github.com/pytorch/pytorch/issues/100701
+    #   - https://github.com/pytorch/pytorch/pull/100811
+    @onlyCUDA
+    @ops(
+        foreach_unary_op_db + foreach_binary_op_db + foreach_pointwise_op_db + foreach_reduce_op_db + foreach_lerp_op_db,
+        dtypes=(torch.float32,)
+    )
+    def test_zero_size_tensor_inputs(self, device, dtype, op):
+        wrapped_op, ref, inplace_op, inplace_ref = self._get_funcs(op)
+
+        for sample in op.sample_zero_size_inputs(device, dtype):
+            if not op.has_no_out_of_place:
+                wrapped_op((sample.input, *sample.args), is_cuda=self.is_cuda, is_fastpath=True, zero_size=True)
+            with InplaceForeachVersionBumpCheck(self, sample.input):
+                inplace_op((sample.input, *sample.args), is_cuda=self.is_cuda, is_fastpath=True, zero_size=True)
+
+    @ops(
+        foreach_unary_op_db + foreach_binary_op_db + foreach_pointwise_op_db + foreach_reduce_op_db + foreach_lerp_op_db,
+    )
+    @parametrize(
+        "noncontiguous,inplace",
+        [(False, False), (False, True), (True, False), (True, True)],
+        name_fn=lambda x, y: '{}_{}'.format(
+            'fastpath' if x else 'slowpath', 'inplace' if y else 'outplace'
+        )
+    )
+    def test_parity(self, device, dtype, op, noncontiguous, inplace):
+        if inplace:
+            _, _, func, ref = self._get_funcs(op)
+        else:
+            func, ref, _, _ = self._get_funcs(op)
+        for sample in op.sample_inputs(device, dtype, noncontiguous=noncontiguous):
+            kwargs = sample.kwargs
+            disable_fastpath = kwargs.pop("disable_fastpath")
+            expect_fastpath = not (noncontiguous or disable_fastpath)
+            # TODO(crcrpar): push this processing to sample func.
```

Comment:
is this todo getting handled by this pr? or nah? if not, we should make this an issue instead of a code todo haha